src/core/instances.jl

```diff
@@ -292,10 +300,20 @@ function Base.run(mi::ModelInstance, ntimesteps::Int=typemax(Int),
 
     starts = mi.starts
     stops = mi.stops
-    step  = step_size(t)
+    step  = isuniform(t)
+
+    if step != -1
+        comp_clocks = [Clock{Timestep}(start, step, stop) for (start, stop) in zip(starts, stops)]
+    else
+        comp_clocks = Array{Clock{VariableTimestep}}(length(starts))
+        for i = 1:length(starts)
+            start_index = findfirst(t, starts[i])
+            stop_index = findfirst(t, stops[i])
+            years = t[start_index:stop+index]
```

Comment:
`stop+index` looks like a typo for `stop_index`

Reply:
Yep :)